crates/ark/src/interface.rs

```diff
@@ -640,20 +640,34 @@ impl RMain {
         buflen: c_int,
         _hist: c_int,
     ) -> ConsoleResult {
+        let info = Self::prompt_info(prompt);
+        log::trace!("R prompt: {}", info.input_prompt);
+
+        // Upon entering read-console, finalize any debug call text that we were capturing.
+        // At this point, the user can either advance the debugger, causing us to capture
+        // a new expression, or execute arbitrary code, where we will reuse a finalized
+        // debug call text to maintain the debug state.
+        self.dap.finalize_call_text();
```

Comment:
I'm not entirely sure if it matters that this now runs _before_ pending lines are handled 🤔